src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1123,40 +1132,96 @@ export class ResultsViewPageStore {
-            results[i].map(sampleId => {
-                sampleMap[sampleId] = true;
-            });
+    readonly studyToSampleIds = remoteData<{ [studyId: string]: { [sampleId: string]: boolean } }>({
+        await:()=>[
+            this._sampleSpecification
+        ],
+        invoke: async ()=>{
+                const sampleListsToQuery: { studyId: string, sampleListId: string }[] = [];
+                const ret: { [studyId: string]: { [sampleId: string]: boolean } } = {};
+                for (const sampleSpec of this._sampleSpecification.result!) {
+                    if (sampleSpec.sampleId) {
+                        ret[sampleSpec.studyId] = ret[sampleSpec.studyId] || {};
+                        ret[sampleSpec.studyId][sampleSpec.sampleId] = true;
+                    } else if (sampleSpec.sampleListId) {
+                        sampleListsToQuery.push(sampleSpec as { studyId: string, sampleListId: string });
+                    }
+                }
+                const results: string[][] = await Promise.all(sampleListsToQuery.map(spec => {
+                    return client.getAllSampleIdsInSampleListUsingGET({
+                        sampleListId: spec.sampleListId
+                    });
+                }));
+                for (let i = 0; i < results.length; i++) {
+                    ret[sampleListsToQuery[i].studyId] = ret[sampleListsToQuery[i].studyId] || {};
+                    const sampleMap = ret[sampleListsToQuery[i].studyId];
+                    results[i].map(sampleId => {
+                        sampleMap[sampleId] = true;
+                    });
+                }
+                return ret;
         }
-        return ret;
+
     }, {});
 
-    @computed get studyToSampleListId(): { [studyId: string]: string } {
-        return this.samplesSpecification.reduce((map, next) => {
-            if (next.sampleListId) {
-                map[next.studyId] = next.sampleListId;
-            }
-            return map;
-        }, {} as {[studyId: string]: string});
-    }
+    // @computed get studyToSampleListId(): { [studyId: string]: string } {
+    //     return this.samplesSpecification.reduce((map, next) => {
+    //         if (next.sampleListId) {
+    //             map[next.studyId] = next.sampleListId;
+    //         }
+    //         return map;
+    //     }, {} as {[studyId: string]: string});
+    // }
+
+    readonly studyToSampleListId = remoteData<{ [studyId: string]: string }>({
+        await:()=>[
+            this._sampleSpecification
+        ],
+        invoke: async ()=>{
+            return this._sampleSpecification.result!.reduce((map, next) => {
+                if (next.sampleListId) {
+                    map[next.studyId] = next.sampleListId;
+                }
+                return map;
+            }, {} as {[studyId: string]: string});
+        }
+    })
+
+
+    readonly _sampleSpecification = remoteData({
```

Comment:
missing an `s` at the end of `sample`. Also I think this one should be without underscore, the current member should be with an underscore in my opinion, since this is the one that wraps and sanitizes it.